apps/admin-gui/src/app/shared/components/members-candidates-list/members-candidates-list.component.ts

```diff
@@ -83,4 +91,97 @@ export class MembersCandidatesListComponent implements OnChanges, AfterViewInit
       row.member === null ? row.richUser.id + 1 : row.member.id + 1}`;
   }
 
+  getEmail(candidate: Candidate): string {
+    return getCandidateEmail(candidate);
+  }
+
+  getOrganization(candidate: Candidate): string {
+    return getExtSourceNameOrOrganizationColumn(candidate);
+  }
+
+  /**
+   * Gets all logins stored in user attributes
+   *
+   * @return users logins
+   */
+  getLogins(memberCandidate: MemberCandidate): string {
+    if (memberCandidate.richUser) {
+      return this.getLoginsForRichUser(memberCandidate.richUser);
+    }
+    else {
+      let logins = this.getLoginsForCandidate(memberCandidate.candidate);
+      if (logins == null || logins === '') {
+        logins = memberCandidate.candidate.userExtSource.login;
+      }
+      return logins;
+    }
+
+
+	}
+
+	getLoginsForRichUser(user: RichUser): string {
+    let logins = '';
+    for (const userAttribute of user.userAttributes) {
+      if (userAttribute.friendlyName.substring(0, 15) === 'login-namespace') {
+        // process only logins which are not null
+        if (userAttribute.value != null) {
+          // append comma
+          if (logins.length > 0) {
+            logins += ", ";
+          }
+          // parse login namespace
+          const parsedNamespace =  userAttribute.friendlyName.substring(16);
+          logins += parsedNamespace + ": " + userAttribute.value;
+        }
+      }
+    }
+    return logins;
+  }
+
+  getLoginsForCandidate(candidate: Candidate): string {
+    let logins = '';
+    for (const prop in candidate.attributes) {
+      if (candidate.attributes.hasOwnProperty(prop)) {
+        if (prop.indexOf('urn:perun:user:attribute-def:def:login-namespace:') !== -1) {
+          if (candidate.attributes[prop] != null) {
+            if(logins.length > 0){
+              logins += ", ";
+            }
+            // parse login namespace
+            const parsedNamespace = prop.substring(49);
```

Comment:
This looks like a magic number. It would be nice to have it in a constant so one can tell what it represents.

Reply:
Honestly I have no clue what does that mean. I just copied that part from old gui. So maybe ask Pavel, what this is all about.

Reply:
I think it is the length of some attribute's namespace. Don't even say that you copied some code without knowing what it does :smile: